docs/geti_tune/api.md

```diff
@@ -0,0 +1,123 @@
+| `DELETE` | `/api/projects/<id>/dataset/items/<id>`           | -       | -                     | Delete a dataset item                              |
+
+### Annotations
+
+| Method   | Path                                                | Payload         | Return          | Description                               |
+|----------|-----------------------------------------------------|-----------------|-----------------|-------------------------------------------|
+| `GET`    | `/api/projects/<id>/dataset/items/<id>/annotations` | -               | annotation info | Get the annotation/prediction for a media |
+| `POST`   | `/api/projects/<id>/dataset/items/<id>/annotations` | annotation info | annotation info | Annotate a media                          |
+| `DELETE` | `/api/projects/<id>/dataset/items/<id>/annotations` | -               | -               | Delete the annotation for a media         |
+
+### Tags
+
+| Method   | Path                                         | Payload                   | Return       | Description                                 |
+|----------|----------------------------------------------|---------------------------|--------------|---------------------------------------------|
+| `GET`    | `/api/projects/<id>/dataset/items/<id>/tags` | -                         | list of tags | List the tags of a dataset item             |
+| `GET`    | `/api/projects/<id>/dataset/tags`            | -                         | list of tags | List the tags used in the dataset           |
+| `PATCH`  | `/api/projects/<id>/dataset/items/tags`      | items, tags to add/remove | -            | Apply or remove tags from one or more items |
+
+### Views
+
+| Method   | Path                                          | Payload   | Return        | Description                      |
+|----------|-----------------------------------------------|-----------|---------------|----------------------------------|
+| `POST`   | `/api/projects/<id>/dataset/views`            | name      | view info     | Create a new dataset view        |
+| `GET`    | `/api/projects/<id>/dataset/views`            | -         | list of views | List the dataset views           |
+| `GET`    | `/api/projects/<id>/dataset/views/<id>`       | -         | view info     | Get info about a dataset view    |
+| `GET`    | `/api/projects/<id>/dataset/views/<id>/items` | -         | list of items | List the items in a dataset view |
+| `POST`   | `/api/projects/<id>/dataset/views/<id>/items` | items ids | -             | Add items to a dataset view      |
+| `DELETE` | `/api/projects/<id>/dataset/views/<id>/items` | items ids | -             | Remove items from a dataset view |
+| `DELETE` | `/api/projects/<id>/dataset/views/<id>`       | -         | -             | Delete a dataset view            |
+
+## Models
+
+| Method   | Path                                          | Payload                         | Return             | Description                              |
```

Comment:
Do we not allow users to upload their own models? (Either trained from Geti, externally, or from a different project)

Reply:
No, it won't be supported for now. As the name suggests, Geti Tune focuses on models that can be fine-tuned. If we allow uploading models from other applications, then we have to deal with challenges and limitations:

1. Models converted to OpenVINO format can be used for inference, but they can't be fine-tuned
2. The models may have an incompatible input/output layout
3. The `rt_info` section of the model XML may not contain the necessary information about labels